### DISCUSSION TITLE
meshagent on M1 macs

### DISCUSSION THREAD

**Discussion by ghost** [Ylianst/MeshCentral#4975] [General]
I'm using meshagent on my M1 Mac and the agent loses the connection upon reboot.

I know that there is also a macOS binary installer but I cannot find any info on how to install, configure and ensure it runs at every reboot. 

Running the wget command downloads and moved the meshagent binary to the first location in $PATH which is /opt/homebrew/bin/meshagent

Shouldn't this be put straight into the /usr/bin location?

When I try to run it I get:

**sh-3.2# meshagent 

The graphical version of this installer cannot run on this system.
Try installing/updating Zenity, and run again.

You can run the text version from the command line with the following command(s): 
./meshagent -connect
./meshagent -update
./meshagent -uninstall**

I've seen some discussion on the Zenity issue but it doesn't seem to be resolved.

Any suggestions?

**Comment by bbrendon** [comment]
you sure its not the ventura problem that's documented in github?

**Reply by ghost** [reply]
Yes but the issues documented (Zenity) are for Linux, not macOS (yes, I know the underlying OS is linux). 

I did find this. Is this the one you're talking about?

https://github.com/Ylianst/MeshCentral/issues/4849
